### PR TITLE
fix: limit ape_accounts plugin to KeyfileAccounts

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

fixes: #156 

### How I did it

Loop through the `accounts` container rather than all the accounts containers accounts.

### How to verify it

1.) Install another account API plugin
2.) Add an account
3.) Run `ape accounts list` and notice that account does not show up in the list

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by unparalleled-js** [comment]
- [ ] Figure out best way to handle `delete()` when it happens to be an account from another plugin

File: src/ape/options.py [hunk 1]
```diff
@@ -31,3 +35,31 @@ def verbose_option(help=""):
...
         help=help,
     )
+
+
+class Alias(click.Choice):
+    """Wraps ``click.Choice`` to load account aliases for the active project at runtime."""
+
+    name = "alias"
+
+    def __init__(self, account_type: Type[AccountAPI] = KeyfileAccount):
```
Comment (unparalleled-js) [inline src/ape/options.py:45]: Now, different account plugins can import this and initialize it with their own custom type.
This will help make it so you only select the accounts for your type.

A couple PRs from now, I will update the ledger (and trezor) plugins!

File: src/ape/options.py [hunk 2]
```diff
@@ -31,3 +36,31 @@ def verbose_option(help=""):
...
+        # NOTE: we purposely skip the constructor of `Choice`
+        self.case_sensitive = False
+        self._account_type = account_type
+
+    @property
+    def choices(self) -> List[str]:  # type: ignore
+        # NOTE: This is a hack to lazy-load the aliases so CLI invocation works properly
+        return [a.alias for a in accounts.get_accounts_by_type(self._account_type)]
```
Comment (unparalleled-js) [inline src/ape/options.py:54]: Now, the choice aliases and confound to a specific type.
Thus, in the `ape_accounts` plugin, we limit them to `KeyfileAccounts`.

This affords the `delete` command to not be able to delete accounts from other plugins (if they happen to have a `delete()` method implemented).

File: src/ape/options.py [hunk 3]
```diff
@@ -31,3 +36,31 @@ def verbose_option(help=""):
...
+
+
+def _require_non_existing_alias(arg):
```
Comment (unparalleled-js) [inline src/ape/options.py:57]: This callback, however, still checks against all the aliases...

I am not sure how important that it, it _could_ probably still work to have aliases with the same name in separate containers. That seems out of scope though, and not that important...

File: src/ape/options.py [hunk 4]
```diff
@@ -31,3 +36,35 @@ def verbose_option(help=""):
...
+    name = "alias"
+
+    def __init__(self, account_type: Type[AccountAPI] = KeyfileAccount):
+        # NOTE: we purposely skip the constructor of `Choice`
+        self.case_sensitive = False
+        self._account_type = account_type
+
+    @property
+    def choices(self) -> List[str]:  # type: ignore
+        # NOTE: This is a hack to lazy-load the aliases so CLI invocation works properly
```
Comment (unparalleled-js) [inline src/ape/options.py:53]: I am not sure if this comment is still needed...
CLI invocation seems to work fine and this does not seem all that "lazy" as it did before

File: src/ape/options.py (hunk 2 above)
Comment (fubuloubu) [reply src/ape/options.py]: Curious if we might want an `--all` flag to display all the accounts w/ their plugin

File: src/ape/api/address.py [hunk 5]
```diff
@@ -20,6 +20,10 @@ def provider(self) -> ProviderAPI:
 
         return self._provider
 
+    @provider.setter
+    def provider(self, value: ProviderAPI):
+        self._provider = value
```
Comment (fubuloubu) [inline src/ape/api/address.py:25]: Hmm... Wonder if we could just remove the `_provider` field, since we can inject it manually via `a = AddressAPI(); a.provider = ...`

Comment (fubuloubu) [reply src/ape/api/address.py:25]: Or even more generally, make "injectable properties" which are properties that are injected after initialization :thinking:

**Review by fubuloubu — APPROVED** [review]

File: src/ape/api/address.py (hunk 5 above)
Comment (unparalleled-js) [reply src/ape/api/address.py:25]: Interesting!

File: src/ape/options.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/options.py]: I could get behind that.

File: src/ape/api/address.py (hunk 5 above)
Comment (fubuloubu) [reply src/ape/api/address.py:25]: Yeah, have to think of this one more. This property in particular is extremely awkward, and I couldn't figure out how to make it work properly. Same with the managers as properties (which is the source of the issue w/ pydantic refactor)

File: src/ape/options.py (hunk 3 above)
Comment (fubuloubu) [reply src/ape/options.py:60]: no, the aliases are a global construct, so we want to make sure they don't collide

File: src/ape/options.py (hunk 1 above)
Comment (fubuloubu) [reply src/ape/options.py:46]: Can the default be `None` which means "select all"?

Comment (unparalleled-js) [reply src/ape/options.py:46]: Ah, that makes sense! I suppose some non-account API related plugins may want to easily allow selection from all accounts.

Comment (fubuloubu) [reply src/ape/options.py]: Exactly!

File: src/ape/api/address.py (hunk 5 above)
Comment (fubuloubu) [reply src/ape/api/address.py:25]: this should be a new PR, not a blocker on this
